### PR TITLE
ddl: Add some limit for `auto_random`

### LINKED ISSUE #17125: Add Some Limits for auto_random
## Development Task

- Add limit `auto_random` can only defined on `bigint` column
- Add Limit for max `shard_bits` length in `auto_random`: 15

### PR DESCRIPTION
**Pull request by Deardrops**
### What problem does this PR solve?

Issue Number: close #17125 

Problem Summary:

- Add limit `auto_random` can only defined on `bigint` column
- Limit the max `shard_bits` length to 15 in `auto_random`.

### What is changed and how it works?

How it Works:

- Check column type before build `auto_random`.
- Check `autoRandBits <= 15` before build `auto_random`.

### Related changes

- None

### Check List 

Tests 

- Unit test
- Integration test

Side effects

- Performance regression
    - Consumes more CPU
    - Consumes more MEM
- Breaking backward compatibility

### Release note 

- Limits on `auto_random` column: the column type can only be `bigint`, and the max number of  `auto_random` bits is restricted to 15.

### CODE REVIEW

File: executor/ddl_test.go [hunk 1]
```diff
@@ -869,36 +869,29 @@ func (s *testAutoRandomSuite) TestAutoRandomBitsData(c *C) {
 	}
 
 	// Test explicit insert.
-	tk.MustExec("create table t (a tinyint primary key auto_random(2), b int)")
-	for i := 1; i <= 100; i++ {
-		tk.MustExec("insert into t values (?, ?)", i, i)
+	autoRandBitsUpperBound := 2<<47 - 1
+	tk.MustExec("create table t (a bigint primary key auto_random(15), b int)")
+	for i := -10; i < 10; i++ {
+		tk.MustExec(fmt.Sprintf("insert into t values(%d, %d)", i+autoRandBitsUpperBound, i))
```
Comment (AilinKid) [inline executor/ddl_test.go:875]: nice check

File: executor/write.go [hunk 2]
```diff
@@ -214,8 +214,12 @@ func rebaseAutoRandomValue(sctx sessionctx.Context, t table.Table, newData *type
 	if err != nil {
 		return err
 	}
-	shardBits := tableInfo.AutoRandomBits + 1 // sign bit is reserved.
-	recordID = recordID << shardBits >> shardBits
+	if recordID < 0 {
+		return nil
+	}
+	layout := autoid.NewAutoRandomIDLayout(&col.FieldType, tableInfo.AutoRandomBits)
+	// Set bits other than incremental_bits to zero.
```
Comment (AilinKid) [inline executor/write.go:221]:
```suggestion
	// Set bits except incremental_bits to zero.
```

**Comment by Deardrops** [comment]
@tangenta Maybe we need approve this pr firstly?

**Pull request merged**